In [1]:

# IMPORTANT: RUN THIS CELL IN ORDER TO IMPORT YOUR KAGGLE DATA SOURCES
# TO THE CORRECT LOCATION (/kaggle/input) IN YOUR NOTEBOOK,
# THEN FEEL FREE TO DELETE THIS CELL.
# NOTE: THIS NOTEBOOK ENVIRONMENT DIFFERS FROM KAGGLE'S R
# ENVIRONMENT SO THERE MAY BE MISSING LIBRARIES USED BY YOUR
# NOTEBOOK.

DATA_SOURCE_MAPPING = 'fitbit:https%3A%2F%2Fstorage.googleapis.com%2Fkaggle-data-sets%2F1041311%2F7746251%2Fbundle%2Farchive.zip%3FX-Goog-Algorithm%3DGOOG4-RSA-SHA256%26X-Goog-Credential%3Dgcp-kaggle-com%2540kaggle-161607.iam.gserviceaccount.com%252F20250706%252Fauto%252Fstorage%252Fgoog4_request%26X-Goog-Date%3D20250706T103324Z%26X-Goog-Expires%3D259200%26X-Goog-SignedHeaders%3Dhost%26X-Goog-Signature%3D2622460e5dc08b49135c98460066a6258c7add1d216adf87d36486047325915d5ec3ab29d587a35a5f9d7bcec1172578677638d8d01f2ee544bc39bd130460e6746d1312ff65e43b398e1b954031ad4e4de8ddfc8796c7695d3baf5d9c1bb8df56d48ce9a403855373dc2a07e9cccc5fb7168bf66feab094013964f7afdb111569daf581697cec315e41daed28482c9434897236bf1e9cf81ae477a253fdec6e168a5b8c453beae527edbb554aa0ba37d2fb0c0d8ff61a66d743353e0f7e6b9cae81a575fe9180ba7da2444a090748511495a3fc898c02528f545e67b6ce36c6b8a429656948b0eb1474fe6874b28bc5a7ffd8033ec74cc87cd711f79f0bdd80'

KAGGLE_INPUT_PATH = '/kaggle/input'
KAGGLE_WORKING_PATH = '/kaggle/working'

system(paste0('sudo umount ', '/kaggle/input'))
system(paste0('sudo rmdir ', '/kaggle/input'))
system(paste0('sudo mkdir -p -- ', KAGGLE_INPUT_PATH), intern=TRUE)
system(paste0('sudo chmod 777 ', KAGGLE_INPUT_PATH), intern=TRUE)
system(
  paste0('sudo ln -sfn ', KAGGLE_INPUT_PATH,' ',file.path('..', 'input')),
  intern=TRUE)

system(paste0('sudo mkdir -p -- ', KAGGLE_WORKING_PATH), intern=TRUE)
system(paste0('sudo chmod 777 ', KAGGLE_WORKING_PATH), intern=TRUE)
system(
  paste0('sudo ln -sfn ', KAGGLE_WORKING_PATH, ' ', file.path('..', 'working')),
  intern=TRUE)

data_source_mappings = strsplit(DATA_SOURCE_MAPPING, ',')[[1]]
for (data_source_mapping in data_source_mappings) {
    path_and_url = strsplit(data_source_mapping, ':')
    directory = path_and_url[[1]][1]
    download_url = URLdecode(path_and_url[[1]][2])
    filename = sub("\\?.+", "", download_url)
    destination_path = file.path(KAGGLE_INPUT_PATH, directory)
    print(paste0('Downloading and uncompressing: ', directory))
    if (endsWith(filename, '.zip')){
      temp = tempfile(fileext = '.zip')
      download.file(download_url, temp)
      unzip(temp, overwrite = TRUE, exdir = destination_path)
      unlink(temp)
    }
    else{
      temp = tempfile(fileext = '.tar')
      download.file(download_url, temp)
      untar(temp, exdir = destination_path)
      unlink(temp)
    }
    print(paste0('Downloaded and uncompressed: ', directory))
}

print(paste0('Data source import complete'))


character(0)

character(0)

character(0)

character(0)

character(0)

character(0)

[1] "Downloading and uncompressing: fitbit"
[1] "Downloaded and uncompressed: fitbit"
[1] "Data source import complete"


# Case study: Bellabeat

## author: Sergio Cupabán

### 04/06/2024



## INTRODUCTION

This is a Bellabeat data analysis case study by the Google Data Analytics Capstone project. In this case study, I will perform many real-world tasks as a data analyst. To answer the key business questions, I will be following the steps of the data analysis process taught in the course: ask, prepare, process, analyze, share, and act.

### Scenario

Bellabeat is a successful small company, but they have the potential to become a larger player in the global smart device market. Urška Sršen, cofounder and Chief Creative Officer of Bellabeat, believes that analyzing smart device fitness data could help unlock new growth opportunities for the company. As a junior Data Analst, I will focus on one of Bellabeat’s products and analyze smart device data to gain insight into how consumers are using their smart devices. The  discovered insights will help to guide marketing strategy for the company. I will present an analysis to the Bellabeat executive team along with your high-level recommendations for Bellabeat’s marketing strategy.

### Characters

__Urška Sršen:__ Bellabeat’s cofounder and Chief Creative Officer

__Sando Mur:__ Mathematician and Bellabeat’s cofounder; key member of the Bellabeat executive team

__Bellabeat marketing analytics team:__ A team of data analysts responsible for collecting, analyzing, and reporting
data that helps guide Bellabeat’s marketing strategy. You joined this team six months ago and have been busy
learning about Bellabeat’’s mission and business goals — as well as how you, as a junior data analyst, can help
Bellabeat achieve them.
● Products
○ Bellabeat app: The Bellab

### Products of BELLABEAT

__Bellabeat app:__ The Bellabeat app provides users with health data related to their activity, sleep, stress, menstrual cycle, and mindfulness habits. This data can help users better understand their current habits and make healthy decisions. The Bellabeat app connects to their line of smart wellness products.

__Leaf:__ Bellabeat’s classic wellness tracker can be worn as a bracelet, necklace, or clip. The Leaf Tracker connects to the Bellabeat app to track activity, sleep, and stress.

__Time:__ This wellness watch combines the timeless look of a classic timepiece with smart technology to track user activity, sleep, and stress. The Time watch connects to the Bellabeat app to provide you with insights into your daily wellness.

__Spring:__ This is a water bottle that tracks daily water intake using smart technology to ensure that you are appropriately hydrated throughout the day. The Spring bottle connects to the Bellabeat app to track your hydration levels.

__Bellabeat membership:__ Bellabeat also offers a subscription-based membership program for users. Membership gives users 24/7 access to fully personalized guidance on nutrition, activity, sleep, health and beauty, and mindfulness, based on their lifestyle and goals.


## ASK

Keeping in mind the key stakeholders mentioned above, my job is to first ask the right questions to identify what problem I am going to solve. As per business task, I have to analyze smart device usage data to get insights on how consumers use smart devices. All questions should focus on finding trends and using the results to make decisions and influence marketing strategies.

## PREPARE
__Dataset:__ the provided dataset is stored in the Kaggle public domain and is made available through Mobius. The data is organized in a long format.

__Data bias or credibility(ROCCC):__ the dataset shows a sample selection bias as it does not reflect the overall population and we are not sure if the sample (33 users) is representative of the sample as a whole. The dataset is not currentFor the purposes of the study the chosen data is from 12th April ’16 to 12th May ’16 for our analysis. Although data from **12th March ’16 to 11th April '16 ** are cited has null values and is missing important information for daily calories, intensities and steps, the data is original, reliable but not comprehensive and complete.
Perhaps other details should have been considered for the study, such as some demographic variables to see another type of influence on the results, even to think about what type of customers can be classified to drive the marketing strategy.

__Data privacy and integrity:__ this Kaggle data set contains a personal fitness tracker from thirty three Fitbit users consented to the submission of personal tracker data, including minute-level output for physical activity, heart rate, and sleep monitoring. It includes information about daily activity, steps, and heart rate that can be used to explore users’ habits.

The data set contains eighteen csv format tables, for this case only some of those will be considered. Our business task are not the detailed performance of users, therefore, we will be focusing on the datasets related to daily timeframe. This means, the daily activity datasets along with the daily sleep data. Some merging tables are going to be useful for the analysis.
Next, some packages and libraries need to be installed:

In [2]:
library(tidyverse)
library(ggplot2)
library(lubridate)
library(dplyr)
library(janitor)
library(skimr)
library(readr)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.2     ✔ tibble    3.3.0
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.0.4     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


ERROR: Error in library(janitor): there is no package called ‘janitor’


Uploading data:


In [ ]:
daily_activity <- read.csv("/kaggle/input/fitbit/mturkfitbit_export_4.12.16-5.12.16/Fitabase Data 4.12.16-5.12.16/dailyActivity_merged.csv")
hourly_calories <- read.csv("/kaggle/input/fitbit/mturkfitbit_export_4.12.16-5.12.16/Fitabase Data 4.12.16-5.12.16/hourlyCalories_merged.csv")
hourly_intensities <- read.csv("/kaggle/input/fitbit/mturkfitbit_export_4.12.16-5.12.16/Fitabase Data 4.12.16-5.12.16/hourlyIntensities_merged.csv")
sleep_day <- read.csv("/kaggle/input/fitbit/mturkfitbit_export_4.12.16-5.12.16/Fitabase Data 4.12.16-5.12.16/sleepDay_merged.csv")
weight_log_info <- read.csv("/kaggle/input/fitbit/mturkfitbit_export_4.12.16-5.12.16/Fitabase Data 4.12.16-5.12.16/weightLogInfo_merged.csv")


Let's see some details ofthe tables.

In [ ]:
glimpse(daily_activity)
glimpse(hourly_calories)
glimpse(hourly_intensities)
glimpse(sleep_day)
glimpse(weight_log_info)

Using head function:

In [ ]:
head(daily_activity)
head(hourly_calories)
head(hourly_intensities)
head(sleep_day)
head(weight_log_info)

Cleaning data:

In [ ]:
# Remove duplicates from each dataset
daily_activity <- daily_activity %>% distinct()
hourly_calories <- hourly_calories %>% distinct()
hourly_intensities <- hourly_intensities %>% distinct()
sleep_day <- sleep_day %>% distinct()
weight_log_info <- weight_log_info %>% distinct()

Changing Date to right format:

In [ ]:
daily_activity$ActivityDate <- as.Date(daily_activity$ActivityDate, format = "%m/%d/%Y")
daily_activity$day_of_week <- format(daily_activity$ActivityDate, "%A", locale = "English")

hourly_calories$ActivityHour <- mdy_hms(hourly_calories$ActivityHour)
hourly_calories$Time <- format(hourly_calories$ActivityHour, format = "%H:%M%:%S")
hourly_calories$Date <- format(hourly_calories$ActivityHour, format = "%m/%d/%y")

hourly_intensities$ActivityHour <- mdy_hms(hourly_intensities$ActivityHour)
hourly_intensities$Time <- format(hourly_intensities$ActivityHour, format = "%H:%M%:%S")
hourly_intensities$Date <- format(hourly_intensities$ActivityHour, format = "%m/%d/%y")

sleep_day$SleepDay <- mdy_hms(sleep_day$SleepDay )

weight_log_info$Date <- as.Date(weight_log_info$Date, format = "%m/%d/%y")


## PROCESS

We are going to look for insights during manipulation of data can start checking the number of elements or rows in each table:

In [ ]:
# data sample per table
nrow(daily_activity)
nrow(hourly_calories)
nrow(hourly_intensities)
nrow(sleep_day)
nrow(weight_log_info)

As we can see, 'hourly_calories' and 'hourly_intensitie' are the tables with highest sample size, while 'weight_log_info' is the smallest sample size. Now we will check the number of individuals.

In [ ]:
# individuals per table
n_distinct(daily_activity$Id)
n_distinct(hourly_calories$Id)
n_distinct(hourly_intensities$Id)
n_distinct(sleep_day$Id)
n_distinct(weight_log_info$Id)

Now we will see some interesting statistic for our tables.


In [ ]:
daily_activity %>%
  select(TotalSteps,
         TotalDistance,VeryActiveMinutes,
         FairlyActiveMinutes,
         LightlyActiveMinutes,
         SedentaryMinutes,
         Calories) %>%
  summary()

If we observe the central tendency for "VeryActiveMinutes" we could say most of the people spend very little time in physical activity with just 21 minutes on average. Based on the same concept, most of the people have a lot of time as "SedentaryMinutes" almost 17 hours, If we had demographic information we might know more details about that time, for instance, we might say most of time is working and increases bucause of weekends y part of that time is asleep time.


In [ ]:
sleep_day %>%
  select(TotalMinutesAsleep,
         TotalTimeInBed) %>%
  summary()

AS we can see, the people sleep around 7 hours and apparently do not take lot of time to sleep. Let's check a little more...

In [ ]:
sleep_day %>%
  group_by(TotalSleepRecords) %>%
  summarise(
    count = n(),
    mean_total_minutes_asleep = mean(TotalMinutesAsleep),
    mean_total_minutes_inbed = mean(TotalTimeInBed),
    total_minutes_asleep = sum(TotalMinutesAsleep),
    total_time_in_bed = sum(TotalTimeInBed))

We can see just few people have 2 or 3 records, that might be snaps even on weekends, but the people mostly have 1 record.

### Some Visualizations

Let's see some relationships:

In [ ]:
ggplot(data=daily_activity, aes(x=TotalSteps, y=SedentaryMinutes, color= Calories)) + geom_point() + geom_smooth(method = "lm", color="red") + ggtitle("Total Steps and Calories burned vs Sedentary Minutes")

ggplot(data = daily_activity, aes(x = TotalSteps, y = TotalDistance, color = Calories)) + geom_point() +
  geom_smooth(color="orange", se= TRUE)  # Adding a linear trend line without confidence intervals


More total steps less sedentary minutes, the relationship is inversely proportional and the same way with burned calories whereas more steps more distance, this relationships is directly proportional.

Let's see what is happening between Steps and Calories per day.

In [ ]:
daily_average_steps <- daily_activity %>%
  group_by(day_of_week)%>%
  summarise(avg_steps=mean(TotalSteps))

ggplot(daily_average_steps, aes(x=day_of_week, y=avg_steps, fill=day_of_week) ) +
  geom_bar(stat = "identity" ) +
  scale_fill_hue(c = 60) + ggtitle("Average steps per Day")+
  theme(legend.position = "none")

daily_average_calories <- daily_activity %>%
  group_by(day_of_week)%>%
  summarise(avg_cals=mean(Calories))

ggplot(daily_average_calories, aes(x=day_of_week, y=avg_cals, fill=day_of_week) ) +
  geom_bar(stat = "identity" ) +
  scale_fill_brewer(palette = "Set3") +
  ggtitle("Average calories burned per day")+
  theme(legend.position = "none")

Tuesday and Saturday are the day with highest average steps and average calories burned
whereas Sunday and Thursday the lowest. We can observe the strong relationship between Calories and Steps per day.

In [ ]:
ggplot(hourly_intensities, aes(x=Time, y=AverageIntensity)) +
  geom_histogram(stat = "identity", fill='purple') + ggtitle("Intensity Distribution along the day") + theme(axis.text.x = element_text(angle = 90))

Intensity increases in two moments, after 5:00 pm we can suspect is after working and the people take time for some "exercise". In th noon is the other increment, this might be the lunch time or change of work shift or just a break.


In [ ]:
# Select only the columns related to minutes
minutes_activity <- daily_activity[, c("SedentaryMinutes", "LightlyActiveMinutes", "FairlyActiveMinutes", "VeryActiveMinutes")]

# Calculate the mean of each activity type
activity_means <- colMeans(minutes_activity)

# Create a data frame for plotting
activity_df <- data.frame(Activity = names(minutes_activity), Minutes = activity_means)


# Plot pie chart
activity_df$Percent <- activity_df$Minutes / sum(activity_df$Minutes) * 100

pie_chart <- ggplot(activity_df, aes(x = "", y = Percent, fill = Activity)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  geom_text(aes(label = paste0(round(Percent, 1), "%")), position = position_stack(vjust = 0.5)) +  # Add percentage labels
  theme_void() +
  theme(legend.position = "right", plot.title = element_text(hjust = 0.4)) +  # Center the title
  labs(title = "Daily Activity Distribution")

print(pie_chart)

The highest portion of individuals is "Sedentary Minutes", and lowest is "Very Active Minutes". This confirms the results and relations seen above as well, a significant portion of individuals spend their time in sedentary activities (This time should be considered for some strategies

## SHARE

Key Takeaways:

* Daily steps and calories burned show a directly proportional relationship, indicating that the more steps taken, the more calories burned.

* The number of steps taken each day and the intensity are directly proportional relationship. Same kind of relationship exists between Time in bed and Sleep .

* From 12:00 p.m. to 3:00 p.m. and from 5:00 p.m. to 8:00 p.m.(after working probably), individuals are physically active,there is a drop in activity, which may indicate that people are taking lunch (or workbreak) during that time.

* On average, people are most active on Saturdays and Tuesdays. Whereas on Thursday and Sunday people is not too active.

* Most participants spent 80% of their day sedentarily, then they spent 15% of the day in light activity, 1.1% in moderate activity, and 1.7% in heavy activity.

## ACT

Recommendations:

* BellaBeat can set goals and incentives for some number of steps to be done to burn the amount of calories by the user.

* With more details about the users, Bellabeat can design a special program to provide sugested diets, exercise routines and more benefits for they according their interests, time disposition, medical and physical conditions

* Keeping the idea above, Bellabeat can create marketing alliances with other companies related to lifestyle and healthy  to promove and create another products related to the business

* Bellabeat can launch advertising and campaigns in sedentary time slots to get more attention and visibility.

